src/proof/cnf_proof.cpp

```diff
@@ -349,6 +349,40 @@ void CnfProof::collectAssertionsForClauses(const IdToSatClause& clauses,
   }
 }
 
+// Detects whether a clause has x v ~x for some x
+// If so, returns the positive occurence's idx first, then the negative's
+Maybe<std::pair<unsigned, unsigned>> CnfProof::detectTrivialTautology(
+    const prop::SatClause& clause)
+{
+  // a map from a SatVariable to its previous occurence's polarity and location
+  std::map<prop::SatVariable, std::pair<bool, unsigned>> varsToPolsAndIndices;
```

Comment:
It probably doesn't change much in this case but I tend to prefer using an `unordered_map` if the keys don't need to be sorted (`unordered_map` is usually implemented as a hash map and `map` as some sort of tree data structure).

Reply:
:+1: